app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -106,9 +124,8 @@ def stack_response
     @stack_response ||= Terraform::Runner::ResponseAsync.new(options[:terraform_stack_id])
   end
 
-  def decrypt_extra_vars(extra_vars)
-    result = extra_vars.deep_dup
-    result.transform_values! { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }
+  def decrypt_vars(input_vars)
+    input_vars.transform_values! { |val| val.kind_of?(String) ? ManageIQ::Password.try_decrypt(val) : val }
```

Comment:
I thought you wanted to not modify the argument that was passed in, if that is still the case you want `transform_values` instead of `transform_values!` (the difference is the latter modifies the argument that is passed in where the latter creates a new object)